staged-documentation.md

```diff
@@ -0,0 +1,282 @@
+    }
+  }
+]
+```
+
+In the following example we create a component that shows a banner to summer campaign.
+
+First we create a `ControllerComponent` that adds the handlebars component to the page data:
+
+```java
+import com.commercetools.sunrise.common.contexts.UserContext;
+import com.commercetools.sunrise.common.pages.PageData;
+import com.commercetools.sunrise.components.ComponentBean;
+import com.commercetools.sunrise.framework.ControllerComponent;
+import com.commercetools.sunrise.hooks.PageDataHook;
+import io.sphere.sdk.models.Base;
+import play.Configuration;
+
+import javax.inject.Inject;
+import java.util.HashMap;
+import java.util.Map;
+
+public final class SummerCampaignControllerComponent extends Base implements ControllerComponent, PageDataHook {
+
+    @Inject
+    private Configuration configuration;
+    @Inject
+    private UserContext userContext;
+
+    @Override
+    public void acceptPageData(final PageData pageData) {
+        final ComponentBean component = createComponentBean();
+        pageData.getContent().addComponent(component);
+    }
+
+    private ComponentBean createComponentBean() {
+        final ComponentBean component = new ComponentBean();
+        //the template is in conf/templates/components/summercampaign/banner.hbs or of course your theme
+        component.setTemplateName("components/summercampaign/banner");//without .hbs!!!
+        Map<String, Object> data = new HashMap<>();
+        data.put("src", configuration.getString("summercampaign.img.src"));
+        data.put("width", configuration.getString("summercampaign.img.width"));
+        data.put("height", configuration.getString("summercampaign.img.height"));
+        data.put("href", demo.productcatalog.routes.SummerCampaignController.show(userContext.languageTag()).url());
+        component.setComponentData(data);
+        return component;
+    }
+}
+```
+
+You still need to wire the `ControllerComponent` to a controller or multiple controllers.
+Since it make sense to put it to all controllers except for checkout, we need a Guice Module:
+
+```java
+import com.commercetools.sunrise.framework.MultiControllerComponentResolver;
+import com.commercetools.sunrise.framework.MultiControllerComponentResolverBuilder;
+import com.google.inject.AbstractModule;
+import com.google.inject.Provides;
+import demo.productcatalog.SummerCampaignControllerComponent;
+
+public class ComponentsModule extends AbstractModule {
+    @Override
+    protected void configure() {
+    }
+
+    @Provides
+    public MultiControllerComponentResolver multiControllerComponentResolver() {
+        return new MultiControllerComponentResolverBuilder()
+                .add(SummerCampaignControllerComponent.class, controller -> !controller.getFrameworkTags().contains("checkout"))
+                .build();
+    }
+}
+```
+
+The `ComponentsModule` can be reused for multiple components, just add a `add` line for building the `MultiControllerComponentResolver`.
```

Comment:
Now it sounds a bit confusing, but I'm not sure what you want to say. I will just say what I understand:

1) The `ComponentsModule` is a good place to put all bindings related to components. But actually I don't think we would have any normal binding in `configure`, right? We will only have the `MultiControllerComponentResolver` binding (at least in the current state).
2) It would be good to explain somewhere the 2 different ways to register components (per controller vs. in the `MultiControllerComponentResolver`), it was a question that popped up during the training.
3) To avoid repeating the "add", I would use "insert", "put" or separate more both "add"s.

Overall if my assumptions are right, I would propose:
"With this Guice Module we can register any component to multiple controllers with a single line of code. You just need to add your component to the `MultiControllerComponentResolver`, which can be easily achieved with its builder."
I would even put this text before the example.